pkg/identity/service.go

```diff
@@ -0,0 +1,186 @@
+package identity
+
+import (
+	"context"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	pb "github.com/linkerd/linkerd2-proxy-api/go/identity"
+	"github.com/linkerd/linkerd2/pkg/tls"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	// DefaultIssuanceLifetime is the default lifetime of certificates issued by
+	// the identity service.
+	DefaultIssuanceLifetime = 24 * time.Hour
+)
+
+type (
+	// Service implements the gRPC service in terms of a Validator and Issuer.
+	Service struct {
+		v Validator
+		i tls.Issuer
```

Comment:
TIOLI: Would you mind having the `Service` struct have a bit more descriptive field names?

Reply:
Sure. I always feel a bit silly having field names that are just the type names: the code ends up reading as `s.issuer.Issue()` and `s.validator.Validate()` which isn't really any clearer than `s.i.Issue()` and s.v.Validate`, imo... perhaps I should just embed the interfaces?  Either way, i'm not opposed to clarifying it.

Reply:
Ooh, big 👍 to embedding both of these interfaces in the `Service` struct.